components/tools/OmeroJava/test/integration/gateway/TransferFacilityTest.java

```diff
@@ -113,4 +169,39 @@ private void initData() throws Exception {
         rf.close();
         return b;
     }
+
+    @Test
+    public void testStreamImageToServer() throws Exception {
+        String type = "uint8";
+        int x = 64;
+        int y = 64;
+        int c = 3;
+        int z = 3;
+        int t = 3;
+
+        ImageData img = transferFacility.createImage(rootCtx, x, y, c, z, t, type,
+                "StreamedImage", null);
+        for (int ic = 0; ic < c; ic++) {
+            for (int iz = 0; iz < z; iz++) {
+                for (int it = 0; it < t; it++) {
+                    byte[] data = createRandomPlaneData(x, y);
+                    transferFacility.uploadPlane(rootCtx, img.getDefaultPixels().getId(), ic, iz,
+                            it, data);
+                }
+            }
+        }
+
+        transferFacility.closeImage(rootCtx, img.getId());
+    }
+
+    private byte[] createRandomPlaneData(int x, int y) {
+        int size = x * y;
```

Comment:
Thanks, yes, good to know. I always forget about the possibility of an int overflow.